### DISCUSSION TITLE
Is there a method to set window focus programmatically?

### DISCUSSION THREAD

**Discussion by desiwko** [dotnet/Silk.NET#1526] [Q&A]
I spent the last couple days trying to implement ImGui multi-viewport which requires backend support for window creation.  One of the hooks that ImGui requires is the ability to set window focus.  The only related functionality I've come across thus far is the FocusChanged event, but that's obviously for detecting when window focus has changed, not changing it.  Is there something exposed somewhere on IWindow or IView or an extension that provides focus functionality that I'm missing?  If not, is there a technical reason it hasn't been implemented?

**Comment by Arugin** [comment]
Same question here. Same problen with ImGui

**Comment by Perksey** [comment]
It looks like GLFW supports this, so we'd happily accept this as a contribution.
